examples/src/rescue_raps/mod.rs

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+use crate::{Example, ExampleOptions};
+use log::debug;
+use rand_utils::rand_array;
+use std::time::Instant;
+use winterfell::{
+    math::{fields::f128::BaseElement, log2, ExtensionOf, FieldElement},
+    ProofOptions, Prover, StarkProof, Trace, VerifierError,
+};
+
+mod custom_trace_table;
+pub use custom_trace_table::RapTraceTable;
+
+use super::rescue::rescue::{self, STATE_WIDTH};
+
+mod air;
+use air::{PublicInputs, RescueRapsAir};
+
+mod prover;
+use prover::RescueRapsProver;
+
+#[cfg(test)]
+mod tests;
+
+// CONSTANTS
+// ================================================================================================
+
+const CYCLE_LENGTH: usize = 16;
+const NUM_HASH_ROUNDS: usize = 14;
+const TRACE_WIDTH: usize = 4 * 2;
+
+// RESCUE SPLIT HASH CHAIN EXAMPLE
+// ================================================================================================
+
+pub fn get_example(options: ExampleOptions, chain_length: usize) -> Box<dyn Example> {
+    Box::new(RescueRapsExample::new(
+        chain_length,
+        options.to_proof_options(42, 4),
+    ))
+}
+
+pub struct RescueRapsExample {
+    options: ProofOptions,
+    chain_length: usize,
+    seeds: Vec<[BaseElement; 2]>,
+    permuted_seeds: Vec<[BaseElement; 2]>,
+    result: [[BaseElement; STATE_WIDTH]; 2],
```

Comment:
Related to the other comment: I wonder if the `result` should be `[[BaseElement; 2]; 2]`. This way, public inputs would include only the rate portion of the state (rather than the entire state).